third_party/move/move-vm/runtime/src/runtime_type_checks.rs

```diff
@@ -689,7 +776,8 @@ impl RuntimeTypeCheck for FullRuntimeTypeCheck {
             },
             Bytecode::VecPushBack(si) => {
                 let (ty, _) = ty_cache.get_signature_index_type(*si, resolver, ty_args)?;
-                operand_stack.pop_ty()?.paranoid_check_eq(ty)?;
+                // For vector elements use assignability
+                operand_stack.pop_ty()?.paranoid_check_assignable(ty)?;
```

Comment:
You should probably check other places where we were using equality, e.g., below `paranoid_check_is_vec_ref_ty` will call `paranoid_check_eq` on vector element type. It should also be changed to assignability?

Reply:
The comment is confusing, there is no assignability but strict equality for the element type of a vector, since assignability is a shallow type concept which does not propagate over type constructors. However, when you push an element on a vector, the element can be adapted before you put it in. This is like you pass an argument to a function, or store a value to a local: this is where shallow is_assignable applies. For the this reason, we also do not have assignabiliy with check_is_vec_ref_ty. I updated the comment.